test/CadlRanchProjectsNonAzure.Tests/type-property-nullable.cs

```diff
@@ -0,0 +1,248 @@
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_Datetime_patchNull() => Test(async (host) =>
+        {
+            string value = "{ \"requiredProperty\": \"foo\", \"nullableProperty\": null }";
+            var response = await new NullableClient(host, null).GetDatetimeClient().PatchNullAsync(BinaryContent.Create(BinaryData.FromString(value)));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_Duration_getNonNull() => Test(async (host) =>
+        {
+            var response = await new NullableClient(host, null).GetDurationClient().GetNonNullAsync();
+            Assert.AreEqual("foo", response.Value.RequiredProperty);
+            Assert.AreEqual(XmlConvert.ToTimeSpan("P123DT22H14M12.011S"), response.Value.NullableProperty);
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_Duration_getNull() => Test(async (host) =>
+        {
+            var response = await new NullableClient(host, null).GetDurationClient().GetNullAsync();
+            Assert.AreEqual("foo", response.Value.RequiredProperty);
+            Assert.AreEqual(null, response.Value.NullableProperty);
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_Duration_patchNonNull() => Test(async (host) =>
+        {
+            var value = new
+            {
+                requiredProperty = "foo",
+                nullableProperty = "P123DT22H14M12.011S"
+            };
+            var response = await new NullableClient(host, null).GetDurationClient().PatchNonNullAsync(BinaryContent.Create(BinaryData.FromObjectAsJson(value)));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_Duration_patchNull() => Test(async (host) =>
+        {
+            string value = "{ \"requiredProperty\": \"foo\", \"nullableProperty\": null }";
+            var response = await new NullableClient(host, null).GetDurationClient().PatchNullAsync(BinaryContent.Create(BinaryData.FromString(value)));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_CollectionsByte_getNonNull() => Test(async (host) =>
+        {
+            var response = await new NullableClient(host, null).GetCollectionsByteClient().GetNonNullAsync();
+            Assert.AreEqual("foo", response.Value.RequiredProperty);
+            Assert.AreEqual(BinaryData.FromString("hello, world!").ToString(), response.Value.NullableProperty.First().ToString());
+            Assert.AreEqual(BinaryData.FromString("hello, world!").ToString(), response.Value.NullableProperty.Last().ToString());
```

Comment:
Please change to use `BinaryDataAssert.AreEqual` here

Reply:
Also we should assert the collection `response.Value.NullableProperty` has 2 items exactly.
Because if the collection only have one item "hello, world!" this also could succeed

Reply:
fixed